Automated release documentation and pypi distribution

mikeywaites opened the pull request:
<!-- Please describe your pull request here. -->
This PR implements some release automation to support a standard release workflow supported by github actions.

* A release-drafter github action has been added that will automatically generate a formatted github release.  

* A release workflow was also added to automate the distribution of the code to pypi when a release is published.

## References

#56 

<!-- References to relevant GitHub issues and pull requests, esp. upstream and downstream changes -->

## Submitter checklist

- [x] The PR request is well described and justified, including the body and the references
- [x] The PR title represents the desired changelog entry
- [x] The repository's code style is followed (see the contributing guide)
- [x] Test coverage that demonstrates that the change works as expected
- [x] For new features, there's necessary documentation in this pull request or in a subsequent PR to [wiremock.org](https://github.com/wiremock/wiremock.org)

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/wiremock/.github/blob/main/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md in your own repository 
-->

mikeywaites committed:
Set up release automation

Added release-drafter github action which implements the standard
wiremock release template along with a new release action which will
automate to pypi

mikeywaites committed:
Bump release version

mikeywaites committed:
Change release type to published

Dont run automated builds for automated pypi releases

oleg-nenashev commented on .github/release-drafter.yml:
```diff
@@ -0,0 +1,59 @@
+    labels:
+      - localization
+  - title: 🌐 Community-related changes
+    labels:
+      - community
+  - title: 👻 Maintenance
+    labels:
+      - chore
+      - maintenance
+  - title: 🚦 Tests
+    labels:
+      - test
+  - title: ✍ Other changes
+  # Default label used by Dependabot
+  - title: 📦 Dependency updates
+    labels:
+      - dependencies
+    collapse-after: 15
+exclude-labels:
+  - skip-changelog
+  - invalid
+
+template: |
+  <!-- Optional: add a release summary here -->
+  $CHANGES
+
+autolabeler:
+  - label: "documentation"
+    files:
+      - "*.md"
+    branch:
+      - '/docs{0,1}\/.+/'
+  - label: "bug"
+    branch:
+      - '/fix\/.+/'
+    title:
+      - "/fix/i"
+  - label: "enhancement"
+    branch:
+      - '/feature\/.+/'
```
```suggestion
_extends: .github
```

oleg-nenashev replied:
You can use global configuration

mikeywaites replied:
As agreed, we will keep the config for now and look at this optimisation separately

oleg-nenashev commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,26 @@
+        env:
+          PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
+        run: |
+          python -m poetry config pypi-token.pypi $PYPI_TOKEN
```
Should it take version from the release argument/tag instead of the metadata? Otherwise it is a matter of time till it is forgotten

mikeywaites replied:
the version will need to be set in the pyproject.toml.

oleg-nenashev reviewed (approved):
The pipeline is a bit unreliable but it will work

mikeywaites committed:
Set version to 2.4.0-alpha

Setting version to -alpha to allow us to test the
release process without making a commitment to the proper
release of 2.4.0

mikeywaites committed:
Update release action

We are opting to use trusted-publisher feature from pypi
rather than requiring PYPI_TOKEN to be used.
This updated workflow uses pypa/gh-action-pypi-publish@release/v1
marketplace action to handle the publish to pypi
